presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java

```diff
@@ -116,6 +121,148 @@ private KafkaTopicFieldDescription createOneFieldDescription(String name, Type t
         return new KafkaTopicFieldDescription(name, type, mapping, null, dataFormat, null, false);
     }
 
+    @Test(dataProvider = "testRoundTripAllFormatsDataProvider")
+    public void testRoundTripAllFormats(RoundTripTestSetup roundTripTestSetup)
+    {
+        String tableName = roundTripTestSetup.getTableName();
+        String fieldNames = roundTripTestSetup.getFieldNames();
+        String fieldValues = roundTripTestSetup.getFieldValues();
+        String comparableName = roundTripTestSetup.getFieldName("f_bigint");
+        Object comparableValue = roundTripTestSetup.getFieldValue("f_bigint");
+
+        assertUpdate("INSERT into write_test." + tableName + " (" + fieldNames + ") VALUES (" + fieldValues + ")", 1);
+        assertQuery("SELECT " + fieldNames + " FROM write_test." + tableName + " WHERE " + comparableName + " = " + comparableValue, "VALUES (" + fieldValues + ")");
+    }
+
+    @DataProvider(name = "testRoundTripAllFormatsDataProvider")
+    public final Object[][] testRoundTripAllFormatsDataProvider()
+    {
+        return testRoundTripAllFormatsData().stream()
+                .map(roundTripTestSetup -> new Object[] {roundTripTestSetup})
+                .toArray(Object[][]::new);
+    }
+
+    private List<RoundTripTestSetup> testRoundTripAllFormatsData()
+    {
+        return ImmutableList.<TestKafkaIntegrationSmokeTest.RoundTripTestSetup>builder()
+                .build();
+    }
+
+    protected static final class RoundTripTestSetup
+    {
+        private final String tableName;
+        private final List<String> fieldNames;
+        private final List<Object> fieldValues;
+        private final int length;
+
+        public RoundTripTestSetup(String tableName, String fieldName, Object fieldValue)
+        {
+            this(tableName, ImmutableList.of(fieldName), ImmutableList.of(fieldValue));
+        }
+
+        public RoundTripTestSetup(String tableName, List<String> fieldNames, List<Object> fieldValues)
+        {
+            try {
+                checkArgument(fieldNames.size() == fieldValues.size(), "sizes of fieldNames and fieldValues are not equal");
+                this.tableName = requireNonNull(tableName, "tableName is null");
+                this.fieldNames = ImmutableList.copyOf(fieldNames);
+                this.fieldValues = ImmutableList.copyOf(fieldValues);
+                this.length = fieldNames.size();
+            }
+            catch (IllegalArgumentException e) {
```

Comment:
please remove this `try-catch`. `IllegalArgumentException` is already a runtime exception, also it is test code so there is no need to use `PrestoException`